app/controllers/repositories_controller.rb

```diff
@@ -198,12 +198,32 @@ def show
   end
 
   def create
-    @client = Client.new(safe_params)
+    prefix = nil
 
+    @client = Client.new(safe_params)
     authorize! :create, @client
 
-    if @client.save
+    provider_prefix = @client.provider.provider_prefixes.select { |_provider_prefix| _provider_prefix.state == "without-repository" }.first
```

Comment:
@svogt0511  Maybe it is safer to do the prefix lookups ahead of the `Client.new(safe_params)` . that way, you know if there are prefixes available before you create any resource. 

And in the case that no prefixes are available, you can send an error right away without creating anything.